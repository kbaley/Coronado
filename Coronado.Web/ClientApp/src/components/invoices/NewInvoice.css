.new-invoice {
    font-size: 24px;
}