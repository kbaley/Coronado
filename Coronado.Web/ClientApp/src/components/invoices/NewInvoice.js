import React from 'react';
import * as actions from '../../actions/invoiceActions';
import { useDispatch, useSelector } from 'react-redux';
import * as Mousetrap from 'mousetrap';
import { NewIcon } from '../icons/NewIcon';
import InvoiceForm from './InvoiceForm';

export default function NewInvoice() {
  const [ show, setShow ] = React.useState(false);
  const customers = useSelector(state => state.customers);
  const dispatch = useDispatch();

  React.useEffect(() => {
    Mousetrap.bind('n i', showForm);

    return function cleanup() {
      Mousetrap.unbind('n i');
    }
  })

  const showForm = () => {
    setShow(true);
    return false;
  }

  const handleClose = () => {
    setShow(false);
  }

  const saveInvoice = (invoice) => {
    dispatch(actions.createInvoice(invoice));
  }

    return (<span>
        <NewIcon onClick={showForm} />
        <InvoiceForm show={show} onClose={handleClose} onSave={saveInvoice} customers={customers} />
      </span>);
  };
