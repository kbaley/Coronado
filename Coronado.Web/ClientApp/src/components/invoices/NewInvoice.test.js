import React from 'react';
import { NewInvoice } from './NewInvoice';
import renderer from 'react-test-renderer';

describe('NewInvoice tests', () => {
  it('should render properly', () => {
    const tree = renderer
      .create(<NewInvoice />)
    expect(tree.toJSON()).toMatchSnapshot();
  }); 
});