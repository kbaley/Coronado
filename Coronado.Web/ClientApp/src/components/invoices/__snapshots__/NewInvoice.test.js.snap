// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewInvoice tests should render properly 1`] = `
<span>
  <span
    className="new-icon icon new-invoice glyphicon glyphicon-plus-sign"
    onClick={[Function]}
  />
</span>
`;
